docs/bbe-guidelines/bbe-guidelines.md

```diff
@@ -0,0 +1,121 @@
+- [Practices related to the folder structure](#practices-related-to-the-folder-structure)
+
+## General best practices
+
+1. Format the Ballerina source using an IDE Plugin.
+
+2. Always, think about the length of the example. A good example should be short.
+
+3. Make the examples self-contained (i.e., avoid depending on concepts other than the one being illustrated) as much as possible.
+
+4. As a practice, use `ballerina/io` methods in main examples and `ballerina/log` in services examples. Do not use both `io:println` and `log:printInfo` in the same example.
+
+5. It is recommended to limit the maximum character count per line to 80 in .bal files. Otherwise, the lines get automatically wrapped in the [Ballerina website](https://ballerina.io/) reducing the readability as shown below.
+
+    Example;
+
+    <img src="images/line-wrap.png" width="600" height="70" alt="wrapping lines" />
+
+6. All keywords and any other word, which needs to be highlighted should be added within backquotes (e.g., `xml`). Do not use a single quote as it will not get highlighted. 
+
+    Example;
+
+    - In the code:
+
+        <img src="images/backquoted-keywords.png" width="400" height="58" alt="backquoted keywords" />
+
+    - In the Baleerina website:
+
+        <img src="images/keyword-in-bio.png" width="400" height="80" alt="keywords in bio" />
+
+    > **Note:** However, the context should be taken into account when formatting words/phrases with backquotes. For example, although the word "record" is a keyword of the language, it need not be highlighted if you are referring to the record construct of the language as shown below.
+
+    ````
+    // Defines an open record type named Student. The { and } delimiters indicate that in addition to the defined fields,
+    // this record type allows additional fields with pure-typed (i.e., anydata|error) values.
+    // The descriptor record { } is equivalent to record {| (anydata|error)...; |}.
+    type Student record {
+    string name;
+    int age;
+    Grades grades;
+    };
+    ````
+
+7. Remove unused imports in `.bal` files.
+
+8. If a new example is added/deleted, update the [index.json](https://github.com/ballerina-platform/ballerina-lang/blob/master/examples/index.json) file as well.
+
+## Practices related to the code comments
+
+1. Add comments to the code blocks as much as possible by starting them with “//”. They will be moved to the RHS boxes in the [Ballerina website](https://ballerina.io/) as a mechanism to describe the code.
+
+    Example;
+
+    - In the code:
+
+        <img src="images/code-comments.png" width="600" height="80" alt="adding code comments" />
+
+    - In the [Ballerina website](https://ballerina.io/):
+
+        ![comment boxes](images/comment-boxes.png "comment-boxes")
+
+2. Since comments are used in the RHS as a mechanism to describe the code, they should be valid sentences (i.e., start with an upper case letter and end with a full stop etc.).
```

Comment:
```suggestion
2. Since comments are used in the right hand side boxes as a mechanism to describe the code, they should be valid sentences (i.e., start with an upper case letter and end with a full stop etc.).
```